### ISSUE TITLE
Elektra-web 1.3 has troubles with system/elektra

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/libelektra#1867]
TODO IN PREPARATION

**Comment by markus2330** [comment]
Unfortunately, I cannot circumvent this problem: `system/elektra` is always there, even after a `kdb stash`.

For you it should be reproducible by:

```
kdb stash
git clone git@github.com:ElektraInitiative/blobs.git
cd configs/real-world
kdb import user/elektra-dup ini < system_elektra.ini
```

(alternatively, `kdb import system/elektra-dup ini < system_elektra.ini ` should work, too)

And then try to navigate into user/elektra-dup

**Comment by omnidan** [comment]
@markus2330 I think it has to do with one of your plugins, can you try installing it with just `-DTOOLS="web"` (which only installs yajl, resolver, etc) and see if it works? I will try to reproduce it on my system.

**Comment by omnidan** [comment]
@markus2330 I tried importing your `system/elektra` keyset as you described, it is working fine for me. Can you let me know which plugins you compiled it with? Is there any plugin you could think of that could be causing this issue (e.g. dynamically creating keys while querying)?

**Comment by markus2330** [comment]
> I tried importing your system/elektra keyset as you described, it is working fine for me.

So you could navigate into system/elektra/modules/dump?

> Can you let me know which plugins you compiled it with?

The relevant ones are only dump, list, and resolver_fm_hpu_b

> Is there any plugin you could think of that could be causing this issue (e.g. dynamically creating keys while querying)?

There is no difference between dynamically generated keys and keys written from a config file. The export of system/elektra should be identical to what web-ui sees on my system.

Btw. maybe it is necessary to also create some keys in user (another namespace) to actually trigger the bug. I'll investigate.

**Comment by markus2330** [comment]
> Btw. maybe it is necessary to also create some keys in user (another namespace) to actually trigger the bug. I'll investigate.

`system/elektra` and `user/sw/elektra/web/#0/current/instances` are enough to trigger the problems.

Seems like web-ui trys to do: `kdb get system/elektra/modules/dump^Msystem` (with a new-line in the key name?)

I tested with chromium 64.0.3282.119 and FirefoxESR 52.7.3.

Even `start:verbose` of elektrad does not output anything.

**Comment by omnidan** [comment]
@markus2330 have you figured out how to trigger the bug yet? can you let me know your compile flags so I can attempt to reproduce it?

I already built the latest elektra version with dump, list and resolver (and yajl) and it still works for me... It is splitting by `\0` (so newlines shouldn't cause a problem). Maybe this is the bug with `kdb ls -0` not working properly?

**Comment by markus2330** [comment]
> have you figured out how to trigger the bug yet?

Yes triggering is easy, but I do not know how to *not* trigger it.

> can you let me know your compile flags so I can attempt to reproduce it?

I usually run scripts/configure-debian on Debian stretch with all deps installed.

> Maybe this is the bug with kdb ls -0 not working properly?

I think thats it, my installed Elektra on this computer does not support kdb ls -0 yet. I'll update...

Please support KDB env var and *please* check for Elektra version.

And please test with the data I prepared in configs/real-world in the blob repo (run ./import). And please also run "kdb mount-info".

Will report how it works with updated Elektra, soon.
